src/Components/SignIn.js

```diff
@@ -13,20 +14,63 @@ const validation = Yup.object().shape({
     .required('Must enter a password')
 })
 
+const SIGN_IN = gql`
+  mutation sign_in($data: SigninInput!){
+    signin(data: $data){
+      user{
+        id
+      }
+      jwt
+    }
+  }
+`
+
 export default function SignIn () {
+  const [error, setError] = useState(null)
+
+  const [signin] = useMutation(SIGN_IN, {
+    onCompleted: LoginSuccess,
+    onError: LoginFailed
+  }
+  )
+
+  function LoginSuccess () {
+    return (
+      setError(false)
+    )
+  }
+
+  function LoginFailed () {
+    return (
+      setError(true)
+    )
+  }
+
   return (
     <div className='outer'>
       <div className='middle'>
         <div className='inner'>
           <div className='wrapper fadeInDown'>
             <div id='formContent'>
               <h1>Sign In</h1>
+              {error &&
+                <h3 className='error'>  oh snap! Something went wrong  </h3>}
+              {error === false &&
+                <h3 className='correct'>  Log in successfully </h3>}
               <Formik
                 initialValues={{ username: '', password: '' }}
                 validationSchema={validation}
-                onSubmit={(values, { setSubmitting, resetForm }) => {
+                
+                onSubmit={async(values, { setSubmitting, resetForm }) => {
```

Comment:
Add a space after `async`:
```suggestion
                onSubmit={async (values, { setSubmitting, resetForm }) => {
```